src/utils/EventUtils.ts

```diff
@@ -151,6 +151,16 @@ export enum MessageModerationState {
     SEE_THROUGH_FOR_CURRENT_USER = "SEE_THROUGH_FOR_CURRENT_USER",
 }
 
+// This is lazily initialized and cached since getMessageModerationState needs it,
+// and is called on timeline rendering hot-paths
+let msc3531Enabled: boolean;
```

Comment:
```suggestion
let msc3531Enabled: boolean | null;
```

or borrow the `Optional<T>` type from the events-sdk